src/sources/file.rs

```diff
@@ -579,16 +582,24 @@ pub fn file_source(
 
         let mut encoding_decoder = encoding_charset.map(Decoder::new);
 
+        let mut file_id_to_metrics_mapping: HashMap<FileFingerprint, FileBytesReceivedHandle> =
+            HashMap::new();
```

Comment:
Two notes after looking at this again:
1. The generic name for the handle is `Registered<FileBytesReceived>`. This is a simple type alias that avoids needing to import the actual name of the handle.
2. Could this table hold both of the handles? i.e. `HashMap<FileFingerprint, (Registered<FileBytesReceived>, Registered<FileEventsReceived>)>` That way, when we need to garbage collect, there is only one table to manage.

Reply:
1. Fixed.
2. I divided them into separate maps due to "use-after-move" issues. As you see, both maps are used in different lambdas with `move` keyword. Of course, we could mitigate it as usual with `Rc`/`Arc` but I didn't want to do so if I could just divide maps and avoid this problem. If there is a better way to merge the maps into one and still have maintainable code and not fight with the borrow checker - please tell me.